### PR TITLE
Made pic-configure's back-end argument independent from compiler

### PR DESCRIPTION
**Pull request by theZiz**
Although most compilers these days understands gcc's way of giving
architecture hints (-march=<architecture> -mtune=<architecture>)
this approach does not work for every compiler (like xlc or gcc
on power8 systems) and not for every case, especially of not using
"native" e.g. for crosscompiling. The Intel compiler e.g. can also
cross compile for the KNL using -xmic-avx512 (-march does not work
here). So I added a function, which checks for the used C++
compiler but also the underlying machine (for power8) to choose
the right option to pass the architecture.

### CODE REVIEW

**Comment by ax3l** [comment]
Thx, that's great! 

I wonder if we should do this in a shell script in the first place and not pass it similar to fast math, as an extra cmake option which then extends the flags. 

The problem is that compilers can have version and arch suffixes, renamings, absolute paths, etc in their name/$CXX and only the cmake compiler id seems as a useful switch to me in terms of side effect reduction and maintainability.

**Comment by theZiz** [comment]
I also see your problem, but I don't think architecture decisions should be part of the CMake file. Maybe making the script based on `$CXX --version | head -n 1` is more realiable. I just tested this for gcc, intel, clang, nvcc and xlc++ and every compiler gave usefull and unambiguous informations. I guess CMake does basically the same. ;)

**Comment by ax3l** [comment]
That's certainly more robust. But please be aware that `$CC` & `$CXX` do not necessarily need to be set as environment variables to have a valid compiler selected by CMake (and many user do not have this var set, e.g. on their desktops). This must work as well.

Nevertheless, I consider the following aspects when tending to put the logic rather in CMake:
- we don't need to maintain compiler detection weirdness in corner cases, CMake does
- the picongpu source code should be considered read-only for users (e.g. system wide package/module). in case things break or need a work-around, setting a cmake option can be done by the users or even hidden as options in profiles, changing the logic inside `pic-configure` script not so. (or course, bugs in logic of CMake scripts have the same shortcome)

we can also post-pone moving that (not) to CMake for later, just my thoughts on this.

File: pic-configure [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #
 
 this_dir=$(cd $(dirname $0) && pwd)
-
+known_cxx=( 'icpc' 'xlc++' 'clang++' 'g++' )
```
[the remark this answers is not in the document]
Comment (theZiz) [reply pic-configure]: Will add it

[the remark this answers is not in the document]
Comment (theZiz) [reply pic-configure]: Does the PGI compiler also use `-march` and `-mtune`?

[the remark this answers is not in the document]
Comment (ax3l) [reply pic-configure]:
```
       -tp=target
              Specify the type of the target processor; possibilities are

              -tp=k8  AMD Opteron or Athlon-64

              -tp=barcelona
                      AMD Barcelona processor

              -tp=shanghai
                      AMD Shanghai architecture Opteron processor

              -tp=istanbul
                      AMD Istanbul architecture Opteron processor

              -tp=bulldozer
                      AMD Bulldozer processor

              -tp=piledriver
                      AMD Piledriver architecture Opteron processor

              -tp=p7  Intel 64 processor

              -tp=core2
                      Intel core2 processor

              -tp=penryn
                      Intel Penryn architecture Pentium processor

              -tp=nehalem
                      Intel Nehalem architecture Core processor

              -tp=sandybridge
                      Intel SandyBridge architecture Core processor

              -tp=haswell
                      Intel Haswell architecture processor

              -tp=px  Blended code generation that will work on any x86-compatible
                      processor

              -tp=x64 Equivalent to -tp=k8,p7.

              The default in the absence of the -tp flag is to compile for the type
              of CPU on which the compiler is running.  Where available,
              -tp=target-64 is equivalent to -m64 -tp=target, and -tp=target-32 is
              equivalent to -m32 -tp=target.  When 32- and 64-bit targets are
              available for a target, -tp=target by itself will compile for a 32-bit
              or 64-bit target depending on whether the 32-bit or 64-bit compiler is
              invoked from your command line path.
```

(`-ta` for the accelerator for OpenACC)

[the remark this answers is not in the document]
Comment (ax3l) [reply pic-configure]: we also have pgi modules on Hypnos

[the remark this answers is not in the document]
Comment (theZiz) [reply pic-configure]: I added support for the PGI compiler. If the PGI compiler is used and the target is set to `native`, I don't define any `-tp` flag to enable compiling for the underlying machine.
`-ta` I ignore for now until Alpaka has working and tested OpenAcc support.

**Comment by ax3l** [comment]
very useful collection! now included in #2296

**Pull request closed without merging**